### ISSUE TITLE
Regression: Support for Context Manger as a decorator is broken since 1.1.394

### ISSUE THREAD

**Issue by benglewis** [microsoft/pyright#9963]
Using a Python decorator as a Context Manger is broken in 1.1.394. In version 1.1.393 it was working.

**Describe the bug**
Here's a couple small examples:

Without type annotations:
```python
from contextlib import contextmanager


@contextmanager
def foo_context():
    yield


class Ben:
    @foo_context()
    def bar(self, b) -> None:
        pass


ben = Ben()
ben.bar(1)
```

With type annotations:
```python
from collections.abc import Iterator
from contextlib import contextmanager


@contextmanager
def foo_context() -> Iterator[None]:
    yield


class Ben:
    @foo_context()
    def bar(self, b: int) -> None:
        print(b)


ben = Ben()
ben.bar(1)
```

The error returned with `pyright` latest (version 1.1.394 currently) is:

> ben.py:17:1 - error: Argument missing for parameter "b" (reportCallIssue)

With version 1.1.393 it returns:
> 0 errors, 0 warnings, 0 informations

I ran the code using the `pyright` CLI directly.

Also note that `pyright` incorrectly complains that it is not the latest version even when it is:

```sh
PYRIGHT_PYTHON_FORCE_VERSION=1.1.394 pyright ben.py
```

> WARNING: there is a new pyright version available (v1.1.394 -> v1.1.394).
> Please install the new version or set PYRIGHT_PYTHON_FORCE_VERSION to `latest`

**Comment by erictraut** [comment]
This regression was introduced by a recent change to the `contextlib.pyi` typeshed stub. It wasn't due to any change in pyright's logic. I recommend reporting the issue to the maintainers of typeshed. This recent change makes assumptions about ParamSpec behaviors that are not specified in the typing spec and should not be relied upon.

As for the "latest version" warning, that is not generated by pyright itself but rather by the community-maintained [Python wrapper for pyright](https://github.com/RobertCraigie/pyright-python). If you think there's a bug in this component, please report the issue to its maintainers.

**Comment by erictraut** [comment]
Here's [a link](https://github.com/microsoft/pyright/commit/ccdf54703988f174357ef3a28c58d1933b7f51b7#diff-8335e334bf827f7cc2e59270b2d64c30610084aca022466369086a74cae6215bR73) to the specific typeshed change that results in this broken behavior.

**Comment by erictraut** [comment]
It looks like the regression has already been reported to the typeshed maintainers, and they've reverted the change [here](https://github.com/python/typeshed/commit/3f41b30535ad614701a47e47659a5d7c89a6bbe8). That means this problem will be fixed when I update the typeshed stubs to the latest version. I typically do this every couple of weeks, but I can expedite it to make sure that this fix makes it into the next release of pyright.

**Comment by erictraut** [comment]
I've updated the typeshed stubs bundled with pyright, and I confirmed that the regression was fixed. This will be included in the next release of pyright.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.395.
